### DISCUSSION TITLE
Obisidian.nvim for every markdown files

### DISCUSSION THREAD

**Discussion by anotherhadi** [nix-community/nixvim#1763] [Q&A]
In the obsidian.nvim plugin, we can choose to enable obsidian.nvim for every markdown file (and not only the vault)
  How could I achieve the same result with nixvim?

Obsidian.nvim doc:
```lua
  ft = "markdown",
  -- Replace the above line with this if you only want to load obsidian.nvim for markdown files in your vault:
  -- event = {
  --   -- If you want to use the home shortcut '~' here you need to call 'vim.fn.expand'.
  --   -- E.g. "BufReadPre " .. vim.fn.expand "~" .. "/my-vault/**.md"
  --   "BufReadPre path/to/my-vault/**.md",
  --   "BufNewFile path/to/my-vault/**.md",
  -- },
  ```

**Comment by MattSturgeon** [comment]
Anything documented upstream should be possible via the [`plugins.obsidian.settings`](https://nix-community.github.io/nixvim/plugins/obsidian/settings/index.html) option (even if we don't explicitly have a corresponding option; settings itself is type `attrsOf anything`).

**Reply by anotherhadi** [reply]
This is not an option but a lazyvim setting, I have no idea how to implement this in Nixvim

**Reply by MattSturgeon** [reply]
Ah, you can do this with the [`files`](https://nix-community.github.io/nixvim/NeovimOptions/index.html#files) option (not to be confused with the `extraFiles` option).

You can nest nixvim config under `files."ftplugins/markdown.lua"` and this will apply that config as a filetype plugin instead of `init.lua`.

e.g.

```nix
{
  files."ftplugins/markdown.lua" = {
    plugins.obsidian.enable = true;
  };
}
```

**Reply by anotherhadi** [reply]
It does only enable obsidian for markdown files so it "should" work.. but I don't have the UI element like in a standard nvim config :(
(Note: ftplugin/markdown.lua and not ftpluginS)
